aten/src/ATen/native/native_functions.yaml

```diff
@@ -3117,15 +3117,11 @@
 
 - func: eq_(Tensor(a!) self, Scalar other) -> Tensor(a!)
   variants: method
-  dispatch:
-    CPU: legacy::cpu::_th_eq_
-    CUDA: legacy::cuda::_th_eq_
+  requires_tensor: True
```

Comment:
@yf225 
For strategy: use_derived to be selected, one of the three has to be true:
- requires_tensor: True is specified
- function is "abstract"
- there is a derivatve specified.

There isn't a derivative specified because this isn't differentiable. The function is not abstract (but I could make it abstract as a solution by setting dispatch to be CPU and CUDA only, I might do that...). The last option available was to set requires_tensor: True

Reply:
In the original code, the function was "abstract" because of the dispatching.

Reply:
For context: we need to select `use_derived` dispatch strategy because we want the returned tensor to be wrapped as a Variable (i.e. calling `as_variable(...)`). For this case, I think using `requires_tensor: True` is okay, because once https://github.com/pytorch/pytorch/issues/23032 is done, we can trivially remove `requires_tensor: True` here since the tensor returned will already be a Variable.